test/optimization/test_admm_miskp.py

```diff
@@ -1,199 +1,95 @@
-
-    # A_2 z + A_3 u \leq b_2 -- > - y_k <= x_{k,t}
-    A2 = np.zeros((m, n))
-    for i in range(K):
-        A2[T * i:T * (i + 1), i] = - np.ones(T)
-    A3 = np.eye(m)
-    b2 = np.zeros(m)
-
-    # A_4 u <= b_3 --> sum_k sum_t D_{kt} x_{kt} \leq P, -x_{kt} \leq 0
-    block1 = D.reshape((1, m))
-    block2 = -np.eye(m)
-    A4 = np.block([[block1], [block2]])
-    b3 = np.hstack([P, np.zeros(m)])
-
-    __dump_arrays(Q0, Q1, c0, c1, A0, b0, A1, b1, A4, b3, A2, A3, b2)
-
-    return K, T, P, S, D, C
+class TestADMMOptimizerMiskp(QiskitOptimizationTestCase):
+    """ADMM Optimizer Tests based on Mixed-Integer Setup Knapsack Problem"""
 
+    def setUp(self):
+        super().setUp()
 
+    def test_admm_optimizer_miskp_eigen(self):
+        """ ADMM Optimizer Test based on Mixed-Integer Setup Knapsack Problem using NumPy eigen optimizer"""
+        K, T, P, S, D, C = self.get_problem_params()
+        miskp = Miskp(K, T, P, S, D, C)
+        op: OptimizationProblem = miskp.create_problem()
 
-class Miskp:
+        # use numpy exact diagonalization
+        qubo_optimizer = MinimumEigenOptimizer(NumPyMinimumEigensolver())
+        continuous_optimizer = CplexOptimizer()
+
+        admm_params = ADMMParameters(qubo_optimizer=qubo_optimizer, continuous_optimizer=continuous_optimizer)
+
+        solver = ADMMOptimizer(params=admm_params)
+        solution = solver.solve(op)
+
+        # debug
+        print("results")
+        print("x={}".format(solution.x))
+        print("fval={}".format(solution.fval))
+
+        correct_solution = [0.009127, 0.009127, 0.009127, 0.009127, 0.009127, 0.009127, 0.009127, 0.009127,
+                            0.009127, 0.009127, 0.006151, 0.006151, 0.006151, 0.006151, 0.006151, 0.006151,
+                            0.006151, 0.006151, 0.006151, 0.006151, 0.,       0.]
+        correct_objective = -1.2113693
+
+        np.testing.assert_almost_equal(correct_solution, solution.x, 3)
+        np.testing.assert_almost_equal(solution.fval, correct_objective, 3)
+
+    def get_problem_params(self):
+        """
+        Fills in parameters for a Mixed Integer Setup Knapsack Problem (MISKP) instance.
+        """
+        #
+
+        K = 2
+        T = 10
+        P = 45.10
+        S = np.asarray([75.61, 75.54])
+
+        D = np.asarray([9.78, 8.81, 9.08, 2.03, 8.9, 9, 4.12, 8.16, 6.55, 4.84, 3.78, 5.13,
+                        4.72, 7.6, 2.83, 1.44, 2.45, 2.24, 6.3, 5.02]).reshape((K, T))
+
+        C = np.asarray([-11.78, -10.81, -11.08, -4.03, -10.90, -11.00, -6.12, -10.16, -8.55, -6.84,
+                        -5.78, -7.13, -6.72, -9.60, -4.83, -3.44, -4.45, -4.24, -8.30, -7.02]).reshape((K, T))
+
+        return K, T, P, S, D, C
 
-    def __init__(self, K, T, P, S, D: np.ndarray, C: np.ndarray, verbose=False, relativeGap=0.0,
-                 pairwise_incomp=0, multiple_choice=0):
+
+class Miskp:
+    def __init__(self, K, T, P, S, D: np.ndarray, C: np.ndarray, pairwise_incomp=0, multiple_choice=0):
```

Comment:
If you use type hints for some args I think you should use them for all 🙂

Reply:
Accepted